### PR TITLE
MULE-18770: Use an expression from configuration element to override default correlationId

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

File: integration/src/test/java/org/mule/shutdown/ValidShutdownTimeoutOneWayTestCase.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
   private TestQueueManager queueManager;
 
   @Rule
-  public SystemProperty contextShutdownTimeout = new SystemProperty("contextShutdownTimeout", "" + RECEIVE_TIMEOUT);
+  public SystemProperty contextShutdownTimeout = new SystemProperty("contextShutdownTimeout", "1" + RECEIVE_TIMEOUT);
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/shutdown/ValidShutdownTimeoutOneWayTestCase.java:43]: ?

File: integration/src/test/resources/org/mule/test/config/correlation-id/static-generation.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns="http://www.mulesoft.org/schema/mule/core"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:http="http://www.mulesoft.org/schema/mule/http"
+      xsi:schemaLocation="
+       http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+       http://www.mulesoft.org/schema/mule/http http://www.mulesoft.org/schema/mule/http/current/mule-http.xsd">
+
+    <configuration correlationIdGeneratorExpression="'doge'"/>
```
Comment (elrodro83) [inline integration/src/test/resources/org/mule/test/config/correlation-id/static-generation.xml:9]: this should fail on deployment

Comment (elrodro83) [reply integration/src/test/resources/org/mule/test/config/correlation-id/static-generation.xml:9]: add a validation for this. Ref: https://github.com/mulesoft/mule/tree/master/runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: integration/src/test/java/org/mule/shutdown/ValidShutdownTimeoutOneWayTestCase.java (hunk 1 above)
Comment (mbuchwald) [reply integration/src/test/java/org/mule/shutdown/ValidShutdownTimeoutOneWayTestCase.java:43]:
```suggestion
  public SystemProperty contextShutdownTimeout = new SystemProperty("contextShutdownTimeout", "" + RECEIVE_TIMEOUT);
```
Idk where this came from...

File: integration/src/test/resources/org/mule/test/config/correlation-id/static-generation.xml (hunk 2 above)
Comment (mbuchwald) [reply integration/src/test/resources/org/mule/test/config/correlation-id/static-generation.xml:9]: Isn't it a valid expression?

**Review by elrodro83 — DISMISSED** [review]

**Review by elrodro83 — APPROVED** [review]

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**